src/allocators/lib/ERC7683AllocatorLib.sol

```diff
@@ -0,0 +1,295 @@
+    //          Order(address arbiter,Lock[] commitments,Mandate mandate)")
+    bytes32 public constant ORDERDATA_GASLESS_TYPEHASH =
+        0xca948fccb29bd545dea3361927ce0b7d8b680a214439e24af475831131515b4c;
+
+    error InvalidRecipientCallbackLength();
+    error InvalidOrderDataType(bytes32 orderDataType, bytes32 expectedOrderDataType);
+    error InvalidOriginSettler(address originSettler, address expectedOriginSettler);
+    error InvalidOrderData(bytes orderData);
+
+    function openForPreparation(IOriginSettler.GaslessCrossChainOrder calldata order, bytes calldata sponsorSignature)
+        internal
+        view
+        returns (
+            IERC7683Allocator.Order calldata orderData,
+            uint32 deposit,
+            bytes32 mandateHash,
+            IOriginSettler.ResolvedCrossChainOrder memory resolvedOrder
+        )
+    {
+        // Check if orderDataType is the one expected by the allocator
+        if (order.orderDataType != ORDERDATA_GASLESS_TYPEHASH) {
+            revert InvalidOrderDataType(order.orderDataType, ORDERDATA_GASLESS_TYPEHASH);
+        }
+        // Check if the originSettler is the allocator
+        if (order.originSettler != address(this)) {
+            revert InvalidOriginSettler(order.originSettler, address(this));
+        }
+
+        // Decode the orderData
+        (orderData, deposit) = decodeOrderData(order.orderData);
+        deposit = sanitizeBool(deposit);
+
+        // Ensure a valid mandate is provided
+        if (orderData.mandate.fills.length == 0 || order.fillDeadline != orderData.mandate.fills[0].expires) {
+            revert InvalidOrderData(order.orderData);
+        }
+        bytes32[] memory fillHashes;
+        (mandateHash, fillHashes) = hashMandate(orderData.mandate);
+        resolvedOrder =
+            resolveOrder(order.user, order.nonce, order.openDeadline, fillHashes, orderData, sponsorSignature);
+    }
+
+    function openPreparation(IOriginSettler.OnchainCrossChainOrder calldata order)
+        internal
+        pure
+        returns (
+            IERC7683Allocator.Order calldata orderData,
+            uint32 expires,
+            bytes32 mandateHash,
+            bytes32[] memory fillHashes
+        )
+    {
+        // Check if orderDataType is the one expected by the allocator
+        if (order.orderDataType != ORDERDATA_ONCHAIN_TYPEHASH) {
+            revert InvalidOrderDataType(order.orderDataType, ORDERDATA_ONCHAIN_TYPEHASH);
+        }
+
+        // Decode the orderData
+        (orderData, expires) = decodeOrderData(order.orderData);
+        expires = sanitizeUint32(expires);
+
+        // Ensure a valid mandate is provided
+        if (orderData.mandate.fills.length == 0 || order.fillDeadline != orderData.mandate.fills[0].expires) {
+            revert InvalidOrderData(order.orderData);
+        }
+
+        (mandateHash, fillHashes) = hashMandate(orderData.mandate);
+    }
+
+    function decodeOrderData(bytes calldata orderData)
```

Comment:
we should make sure we have natspec on anything that doesn't inherit it from an interface